src/main/scala/io/iohk/ethereum/crypto/ECDSASignature.scala

```diff
@@ -111,6 +111,12 @@ object ECDSASignature {
     curve.getCurve.decodePoint(compEnc)
   }
 
+  def hasOnlyLastByteSet(v: ByteString): Boolean = {
```

Comment:
How about `v.dropWhile(_ == 0).size == 1`? No `if`.
Does this function have anything to do `ECDSASignature`?

Reply:
cool one liner definitely an improvement over `if`. 
As for reason of putting it in `ECDSASignature` , theoretically, we could use this check in many contexts, but for now, we are using it only in the context of checking `ECDSASignature` recovery id.

Reply:
But it doesn't belong in this class/object. As a new user of this class I would have no idea why I might need this method. And even if you were to explain it in scaladoc, it's still very poor design to require a 2-step interaction with this class.

The primary problem here is that the API of `ECDSASignature` is wrong. It tells us it accepts any `ByteString` for the value of `v` but it does not handle it correctly - and this has been the source of the bug in the precompiled contract. We need to change this API so that either:
- it accepts `ByteString`, but handles _any possible value_ correctly, returning an error for invalid `v`
- it only accepts `Byte` type for `v`, and this should be already handled correctly

The latter should be easier.